File: echoUtils/src/main/java/com/echo/library/InvokeManager.java
package com.echo.library;


import com.echo.library.util.CommonUtils;

import java.util.HashMap;

/**
 * author   : dongjunjie.mail@qq.com
 * time     : 2021/6/29
 * change   :
 * describe :
 */
public class InvokeManager {
    //请求最小间隔
    public static final int MinimumIntervalTime = 500;

    static final HashMap<String, Long> map = new HashMap<>();

    public static boolean canInvoke() {
        StackTraceElement[] traceElement = new Throwable().getStackTrace();
        String invokeMethod = traceElement[1].toString();
        Long lastInvokeTime = map.get(invokeMethod);
        if (lastInvokeTime == null) {
            lastInvokeTime = 0L;
        }
        long now = System.currentTimeMillis();
        long intervalTime = now - lastInvokeTime;
        CommonUtils.log(invokeMethod, "lastInvokeTime", lastInvokeTime, "intervalTime", intervalTime);
        if (intervalTime > MinimumIntervalTime) {
            map.put(invokeMethod, now);
            return true;
        }
        return false;
    }
}

File: echoUtils/src/main/java/com/echo/library/InvokeManager.kt
package com.echo.library

import com.echo.library.util.CommonUtils
import java.util.*

/**
 * author   : dongjunjie.mail@qq.com
 * time     : 2021/6/29
 * change   :
 * describe :
 */
object InvokeManager {
    //请求最小间隔
    private const val MinimumIntervalTime = 500
    private val map = HashMap<String, Long>()
    fun canInvoke(): Boolean {
        val traceElement = Throwable().stackTrace
        val invokeMethod = traceElement[1].toString()
        var lastInvokeTime = map[invokeMethod]
        if (lastInvokeTime == null) {
            lastInvokeTime = 0L
        }
        val now = System.currentTimeMillis()
        val intervalTime = now - lastInvokeTime
        CommonUtils.log(invokeMethod, "lastInvokeTime", lastInvokeTime, "intervalTime", intervalTime)
        if (intervalTime > MinimumIntervalTime) {
            map[invokeMethod] = now
            return true
        }
        return false
    }
}